wgpu-core/src/resource.rs

```diff
@@ -981,6 +983,44 @@ impl Drop for FlushedStagingBuffer {
     }
 }
 
+#[derive(Debug)]
+pub struct ScratchBuffer {
+    raw: ManuallyDrop<Box<dyn hal::DynBuffer>>,
+    device: Arc<Device>,
+}
+
+impl ScratchBuffer {
+    pub(crate) fn new(device: &Arc<Device>, size: wgt::BufferSize) -> Result<Self, DeviceError> {
+        let raw = unsafe {
+            device
+                .raw()
+                .create_buffer(&hal::BufferDescriptor {
+                    label: Some("(wgpu) scratch buffer"),
+                    size: size.get(),
+                    usage: BufferUses::ACCELERATION_STRUCTURE_SCRATCH | BufferUses::MAP_WRITE,
+                    memory_flags: hal::MemoryFlags::empty(),
+                })
+                .map_err(crate::device::DeviceError::from_hal)?
+        };
+        Ok(Self {
+            raw: ManuallyDrop::new(raw),
+            device: device.clone(),
+        })
+    }
+    pub(crate) fn raw(&self) -> &dyn hal::DynBuffer {
+        self.raw.as_ref()
+    }
+}
+
+impl Drop for ScratchBuffer {
+    fn drop(&mut self) {
+        resource_log!("Destroy raw StagingBuffer");
+        // SAFETY: We are in the Drop impl and we don't use self.raw anymore after this point.
+        let raw = unsafe { ManuallyDrop::take(&mut self.raw) };
+        unsafe { self.device.raw().destroy_buffer(raw) };
+    }
+}
+
```

Comment:
Could we plop this in its own module? In generally trying to pare down module sizes, if possible.

Reply:
Alternatively we could put it in the pre-existing ray_tracing.rs, it feels a little small to be on it's own.

Reply:
Honestly, I'd rather too small than too large at this point - it's not really inherently related to rt, just a nice way to get a scratch buffer